pages/hold/request/[id].tsx

```diff
@@ -0,0 +1,240 @@
+import { fetchBib } from "../../../src/server/api/bib"
+import initializePatronTokenAuth, {
+  doRedirectBasedOnNyplAccountRedirects,
+  getLoginRedirect,
+} from "../../../src/server/auth"
+import { getPatronData } from "../../api/account/[id]"
+
+import Bib from "../../../src/models/Bib"
+import Item from "../../../src/models/Item"
+
+import bibDetailStyles from "../../../styles/components/BibDetails.module.scss"
+
+import type { DiscoveryBibResult } from "../../../src/types/bibTypes"
+import type { DiscoveryItemResult } from "../../../src/types/itemTypes"
+
+interface HoldRequestPropsType {
+  discoveryBibResult: DiscoveryBibResult
+  discoveryItemResult: DiscoveryItemResult
+  isAuthenticated?: boolean
+}
+
+/**
+ * The Hold Request page renders a form to place a hold on an item.
+ * It renders some basic info about the bib and item based on the bib and item ids in the url,
+ * including pickup location options rendered as radio buttons in the form.
+ */
+export default function HoldRequestPage({
+  discoveryBibResult,
+  discoveryItemResult,
+  isAuthenticated,
+}: HoldRequestPropsType) {
+  const metadataTitle = `Item Request | ${SITE_NAME}`
+
+  const bib = new Bib(discoveryBibResult)
+  const item = new Item(discoveryItemResult, bib)
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    console.log("submitting form")
+  }
+
+  return (
+    <>
+      <Head>
+        <meta property="og:title" content={metadataTitle} key="og-title" />
+        <meta
+          property="og:site_name"
+          content={metadataTitle}
+          key="og-site-name"
+        />
+        <meta name="twitter:title" content={metadataTitle} key="tw-title" />
+        <title key="main-title">{metadataTitle}</title>
+      </Head>
+      <Layout isAuthenticated={isAuthenticated} activePage="search">
+        <Heading level="h2" mb="l" size="heading3">
+          Request for on-site use
+        </Heading>
+        <List
+          noStyling
+          type="dl"
+          showRowDividers={false}
+          className={bibDetailStyles.bibDetails}
+          mb="xs"
+          mt={0}
+        >
+          <LinkedDetailElement
+            label="Title"
+            value={[
+              { url: `${PATHS.BIB}/${bib.id}`, urlLabel: bib.titleDisplay },
+            ]}
+            link="internal"
+          />
+          <PlainTextElement label="Call number" value={[item.callNumber]} />
+          {item.volume ? (
+            <PlainTextElement label="Volume/date" value={[item.volume]} />
+          ) : null}
+        </List>
+        <Heading level="h3" size="heading4" mb="l">
+          Choose a pickup location
+        </Heading>
+        <Form
+          id="hold-request-form"
+          data-testid="hold-request-form"
+          // We are using a post request on hold requests when JS is disabled
+          method="post"
+          action={`${BASE_URL}/api/hold/request/${bib.id}-${item.id}`}
```

Comment:
to be clear, this route will be created in a different PR, right?

Reply:
Correct, i can add the skeleton now but will be hard to test before server functions are done.